### DISCUSSION TITLE
how to get dependencies in constructor

### DISCUSSION THREAD

**Discussion by YousefAK009** [kaboc/pot#12] [Q&A] [answered]
How do I get dependencies inside the pot?

**Comment by kaboc** [comment]
Hi @yousefak007. Can you explain in a little more detail what you want to do? A code sample (or an error message if you are getting an error) would be helpful.

**Reply by YousefAK009** [reply]
<img width="681" alt="4c274b4faa6bdd1c9de25005e6be2183" src="https://github.com/kaboc/pot/assets/46021584/ad2662e2-5ca9-4b23-a8f8-32d572876c14">

in get_it I can inject dependencies in the constructor

**Reply by kaboc** [reply]
Pot is used via a global variable, so it is accessible from anywhere. It means constructor injection is not necessary. However, if you still prefer it, it is possible to inject an object held in a pot.

```dart
final fooPot = Pot(() => Foo());
final barPot = Pot(() => Bar(foo: fooPot()));
```

I think it is also possible to pass a pot itself, but it is a bit weird IMO.

```dart
final fooPot = Pot(() => Foo());
final barPot = Pot(() => Bar(fooPot: fooPot));
```

**Reply by YousefAK009** [reply]
aha, so it is the same in pottery I just need to call it?

**Reply by kaboc** [reply, marked as answer]
Yes, exactly.

```dart
Pottery(
  pots: {
    somePot: () => anotherPot(),
  },
  ...
)
```

Hope this helps.

**Reply by YousefAK009** [reply]
thanks
